src/Seed.php

```diff
@@ -423,6 +403,64 @@ public static function unpublishPathAlias($pathAlias) {
     \Drupal::service('event_dispatcher')->dispatch(new QuantEvent('', $alias, [], NULL), QuantEvent::UNPUBLISH);
   }
 
+  /**
+   * Handle canonical redirects.
+   *
+   * Example redirects:
+   * - en node 123, no alias: /node/123 to /en/node/123.
+   * - es node 123, no alias: /node/123 to /es/node/123.
+   * - en node 123, alias: /node/123 and /en/node/123 to en alias.
+   * - es node 123, alias: /node/123 and /es/node/123 to es alias.
+   * - en node 123, es translation, no alias: /node/123 to /en/node/123.
+   * - en node 123, es translation, alias: /node/123 and /en/node/123 to en
+   *   alias, /es/node/123 to es alias.
+   *
+   * @todo Create simpler logic for when multilingual isn't used?
+   */
+  public static function handleCanonicalRedirects($entity, $langcode, $url) {
+    $type = $entity->getEntityTypeId();
+    if (!in_array($type, ['node', 'taxonomy_term'])) {
+      \Drupal::logger('quant_seed')->error('Quant: handleCanonicalRedirects called with wrong type [@type]', ['@type' => $type]);
+      return;
+    }
+
+    $id = $entity->id();
+    $published = $entity->isPublished();
+    $canonical = ($type == 'node') ? "/node/{$id}" : "/taxonomy/term/{$id}";
```

Comment:
Is it better to use `$entity->toUrl('canonical')` ([link](https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Entity%21EntityInterface.php/function/EntityInterface%3A%3AtoUrl/10)) here to generate these based on entity configuration?

Reply:
Good question. Let me do a quick check if that mucks anything switching to that.

Reply:
@steveworley okay... I guess we are using the word "canonical" but it's not really. It's the Drupal path which is called "alias" in `toUrl` which is *very* confusing since it's not the alias! :D

Here's what the results are:

```
    $canonical = ($type == 'node') ? "/node/{$id}" : "/taxonomy/term/{$id}";
    $canonical2 = $entity->toUrl('canonical')->toString();
    $canonical3 = $entity->toUrl('canonical', ['alias'=> TRUE])->toString();
```

```
type = node / canonical = /node/76 / canonical2 = /en/lkasjdflsdfkjgasd / canonical3 = /en/node/76
```

but we don't want the path prefix in this case because we are handling those.

So... 

1. Do we want to update the code to not call these "canonical" since that's at odds with Drupal terminology? e.g. we can use "internal path" or something like that
2. Should we use `toUrl` with `alias` option and strip the path prefix? Or just use our hardcoded approach? If the former, I can add a comment as to why the `alias` option is used
3. Other thoughts?

Reply:
Yeah, let's try and keep terminology aligned with what Core is doing. I'm for calling them internal paths. Much of a muchness, it'll probably require rewriting more to leverage exactly what core is doing with lingual internal paths. So for now I think we just keep our hardcoded logic.